src/kernel/src/process/sleep.rs

```diff
@@ -0,0 +1,95 @@
+use crate::syscalls::{SysErr, SysOut};
+use std::{io::Error, num::NonZeroI32};
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct Timespec {
+    pub seconds: i64,     // tv_sec
+    pub nanoseconds: i64, // tv_nsec
+}
```

Comment:
I think this one should not be here. My idea is create a new module `time` on the top-level and move it to that module.